pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -564,7 +570,7 @@ private PGStream enableGSSEncrypted(PGStream pgStream, GSSEncMode gssEncMode, St
   }
 
   private PGStream enableSSL(PGStream pgStream, SslMode sslMode, Properties info,
-      int connectTimeout)
+      int connectTimeout, SSLNegotiation sslNegotiation)
```

Comment:
Is there a reason to pass `SSLNegotiation` explicitly? It looks like a duplicate
Can the method infer it from `Properties info`?

Reply:
Yes, I suppose it could

Reply:
However the same could be said for SslMode.

Reply:
`SslMode` is slightly different: `org.postgresql.core.v3.ConnectionFactoryImpl#openConnectionImpl` calls `tryConnect` with a changing value of `SslMode` to try all the options.
For instance, if user configures `sslmode=prefer`, then `tryConnect` first attempts with `SslMode.REQUIRE`, and if it fails, then it attempts with `SslMode.DISABLE`.

Do we want to support similar logic for `SSLNegotiation`?
We have no `SSLNegotiation.TRY_DIRECT_AND_FALLBACK_TO_POSTGRESQL`, so inferring the negotiation from `info` is fine.